modules/installation-special-config-kmod.adoc

```diff
@@ -0,0 +1,304 @@
++
+----
+$ mkdir kmods; cd kmods
+$ git clone https://github.com/dustymabe/kmods-via-containers.git
+----
+
+. Install a KVC framework instance on your build host to test the module.
+This adds a kmods-via-container systemd service and loads it:
++
+----
+$ cd kmods-via-containers/
+$ make install
+$ sudo systemctl daemon-reload
+----
+
+. Get the kernel module. This could be a third-party module that you don’t
+have control over, but is supplied by others. You will need configuration files
+and source code similar to those shown in the kvc-simple-kmod example that can
+be cloned to your system as follows:
++
+----
+$ cd ..
+$ git clone https://github.com/dustymabe/kvc-simple-kmod
+----
+
+. Edit the configuration file (`simple-kmod.conf`, in his example) and
+change the name of the Dockerfile to Dockerfile.rhel so the file appears as
+shown here:
++
+----
+$ cd kvc-simple-kmod
+$ cat simple-kmod.conf
+
+KMOD_CONTAINER_BUILD_CONTEXT="git://github.com/dustymabe/kvc-simple-kmod.git"
+KMOD_CONTAINER_BUILD_FILE=Dockerfile.rhel
+KMOD_SOFTWARE_VERSION=dd1a7d4
+KMOD_NAMES="simple-kmod simple-procfs-kmod"
+----
+
+. Create an instance of kmods-via-containers@.service for your kernel module
+(simple-kmod in this example) and enable it:
++
+----
+$ sudo make install
+$ sudo kmods-via-containers build simple-kmod $(uname -r)
+----
+. Enable and start the systemd service, then check the status:
++
+----
+$ sudo systemctl enable kmods-via-containers@simple-kmod.service
+$ sudo systemctl start kmods-via-containers@simple-kmod.service
+$ sudo systemctl status kmods-via-containers@simple-kmod.service
+● kmods-via-containers@simple-kmod.service - Kmods Via Containers - simple-kmod
+   Loaded: loaded (/etc/systemd/system/kmods-via-containers@.service;
+          enabled; vendor preset: disabled)
+   Active: active (exited) since Sun 2020-01-12 23:49:49 EST; 5s ago...
+----
+
+. Here are a few ways to interact with the kernel modules in this example and make sure they were loaded:
++
+----
+$ lsmod | grep simple_
+simple_procfs_kmod     16384  0
+simple_kmod            16384  0
+$ dmesg | grep 'Hello world'
+[ 6420.761332] Hello world from simple_kmod.
+$ sudo cat /proc/simple-procfs-kmod
+simple-procfs-kmod number = 0
+$ spkut 44
+KVC: wrapper simple-kmod for 4.18.0-147.3.1.el8_1.x86_64
+Running userspace wrapper using the kernel module container...
++ podman run -i --rm --privileged
+   simple-kmod-dd1a7d4:4.18.0-147.3.1.el8_1.x86_64 spkut 44
+simple-procfs-kmod number = 0
+simple-procfs-kmod number = 44
+----
+
+Going forward, when the system boots this service will check if a new
+kernel is running. If there is a new kernel, the service builds a new
+version of the kernel module and then loads it. If the module is already
+built, it will just load it.
+
+== Provision the kernel module via Ignition
+Depending on whether or not you need to have the kernel module in place
+when OpenShift Container Platform cluster first boots, you can set up the
```

Comment:
s/OpenShift Container Platform/{product-title}

Reply:
How did that sneak in there? Done.